kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -186,6 +191,69 @@ public LogSegment(
                         Comparator.comparing((FileStatus a) -> new Path(a.getPath()).getName())
                             .reversed())
                     .collect(Collectors.toList()));
+
+    this.compactionsReversed =
```

Comment:
Question: what are the guarantees / expectations for the compactions given to LogSegment?

Should we filter and guarantee that for all compactions, x.y.json, x is >= checkpointVersion + 1 (if present), and y is <= version?

Reply:
Can we also clarify the ordering of the compactions? It is lexicographical ordering, right?

So:

```
0010.0011.json
0010.9999.json
0014.0016.json
0015.0017.json
0015.0019.json
0016.0018.json
```

Reply:
Yep. This is already checked in the `SnapshotManager`, but I've added `checkArgument` validations here.